src/eos-kolibri-import

```diff
@@ -0,0 +1,284 @@
+        shutil.copy(orig_db, kolibri_db)
+
+        logger.info(f'Starting Kolibri server in {kolibri_home}')
+        start_cmd = (
+            'flatpak',
+            'run',
+            '--no-desktop',
+            '--die-with-parent',
+            f'--env=KOLIBRI_HOME={kolibri_home}',
+            f'--env=KOLIBRI_CONTENT_FALLBACK_DIRS={orig_content}',
+            '--env=KOLIBRI_NO_FILE_BASED_LOGGING=true',
+            f'--filesystem={kolibri_home}',
+            f'--filesystem={orig_content}',
+            '--command=/app/bin/kolibri',
+            'org.learningequality.Kolibri',
+            'start',
+            '--foreground',
+        )
+        log_command(start_cmd)
+        self._proc = subprocess.Popen(start_cmd)
+
+        port = self._get_server_port()
+        self.url = f'http://127.0.0.1:{port}'
+        logger.info(f'Kolibri server running at {self.url}')
+
+    def _read_server_port(self):
+        pid_path = os.path.join(self.home.name, 'server.pid')
+        logger.debug(f'Reading Kolibri server port from {pid_path}')
+
+        try:
+            with open(pid_path) as f:
+                lines = f.readlines(1024)
+        except FileNotFoundError:
+            return None
+
+        try:
+            port = int(lines[1].strip())
+        except (IndexError, ValueError):
+            return None
+
+        return port if port != 0 else None
+
+    def _get_server_port(self, timeout=10):
+        deadline = time.time() + timeout
+        while True:
+            port = self._read_server_port()
+            if port:
+                return port
+            if time.time() >= deadline:
+                raise Exception(
+                    f'Could not read server port in {timeout} seconds'
+                )
+            time.sleep(0.5)
+
+    def _stop(self):
+        if not self._proc:
+            return
+
+        # Note that this will not stop the server gracefully since the
+        # process is actually bwrap and it will kill kolibri with
+        # SIGKILL when using flatpak run --die-with-parent. To
+        # gracefully stop, we'd need to send SIGTERM to the process
+        # inside the sandbox. We don't care since this is all running in
+        # a temporary directory.
+        logger.info(f'Stopping Kolibri server {self._proc.pid}')
+        self._proc.terminate()
+        self._proc.wait(timeout=20)
+
+
+def import_channels(kolibri_home, server_url):
+    channels_ini = os.path.join(kolibri_home, 'channels.ini')
+    channels_config = ConfigParser()
+    if not channels_config.read([channels_ini]):
+        raise Exception(f'Could not read {channels_ini}')
+
+    channel_ids = channels_config.get(
+        'kolibri',
+        'install_channels',
+        fallback='',
+    ).split()
+    if not channel_ids:
+        logger.warning(f'No channels configured in {channels_ini}')
+        return
+
+    # Data can't be imported while the server is running.
+    ensure_system_server_stopped()
```

Comment:
Aha, I knew that Stop method would be useful some day! :b One tricky bit here: if the desktop app is running, it'll restart the server right away here, but I don't think there's a significant risk of damage (is there?) so might be worth just leaving a note in the docs or something.

Reply:
Ah, here's where I did that earlier! I forgot we have a whole Python project in this repo do it:

https://github.com/endlessm/eos-kolibri/blob/master/src/eos_kolibri/eos_kolibri_migrate/main.py#L56-L66

It's a bit antiquated, so probably doing more than it needs to at this point.